Change vertical scrolling

reichlfl opened the pull request:
Vertical scrolling is now handled by adjusting the position of
`TimeGraph`. This removes the need for `CaptureViewElement` to
explicitly need to know about scrolling in `TimeGraph`.

Bug: Part of b/176086740 to remove the dependency of `TimeGraph` from `Track`.

karupayun commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -530,8 +535,8 @@ void TimeGraph::PrepareBatcherAndUpdatePrimitives(PickingMode picking_mode) {
 
-  text_renderer_static_.PushTranslation(0, -vertical_scrolling_offset_);
-  batcher_.PushTranslation(0, -vertical_scrolling_offset_);
+  // text_renderer_static_.PushTranslation(0, -vertical_scrolling_offset_);
```
Erase these lines

reichlfl replied:
Thanks, I missed those after reorganizing my commits...

karupayun commented on src/OrbitGl/CaptureViewElement.cpp:
```diff
@@ -124,7 +120,7 @@ std::vector<CaptureViewElement*> CaptureViewElement::GetChildrenVisibleInViewpor
     float child_top_y = child->GetPos()[1];
     float child_bottom_y = child_top_y + child->GetHeight();
-    float screen_top_y = time_graph_->GetVerticalScrollingOffset();
+    float screen_top_y = 0;  // timeline_container_->GetVerticalScrollingOffset();
```
Erase this comment.

karupayun commented on src/OrbitGl/CaptureWindow.cpp:
```diff
@@ -121,8 +121,9 @@ void CaptureWindow::MouseMoved(int x, int y, bool left, bool right, bool middle)
-                                            mouse_click_pos_world_[1] - mouse_pos_world[1]);
+    float new_vertical_offset =
+        timegraph_click_scrolling_offset_ + mouse_click_pos_world_[1] - mouse_pos_world[1];
+    time_graph_->SetAndClampYPos(-new_vertical_offset);
```
Could you add a comment explaining why this is negative?

reichlfl replied:
No longer needed after my update

karupayun commented on src/OrbitGl/TimeGraph.h:
```diff
@@ -93,6 +93,8 @@ class TimeGraph : public orbit_gl::CaptureViewElement {
   void VerticallyMoveIntoView(const orbit_client_protos::TimerInfo& timer_info);
   void VerticallyMoveIntoView(Track& track);
 
+  void SetAndClampYPos(float y);
```
I'm a bit surprised about this method being called using always a negative value of scrolling_offset? Aren't there a way to make them consistently using the same sign?

reichlfl replied:
This has now been removed

florian-kuebler commented on src/OrbitGl/CaptureViewElement.cpp:
```diff
@@ -124,7 +120,7 @@ std::vector<CaptureViewElement*> CaptureViewElement::GetChildrenVisibleInViewpor
     float child_top_y = child->GetPos()[1];
     float child_bottom_y = child_top_y + child->GetHeight();
-    float screen_top_y = time_graph_->GetVerticalScrollingOffset();
+    float screen_top_y = 0;  // timeline_container_->GetVerticalScrollingOffset();
```
Do you want to keep the comment?

reichlfl replied:
> Hi Florian, thanks for this! :)
> 
> I have a question. Did you test what's happening if we set TimeGraph to start at y = 20, instead of 0? I will need this soon because I would like to have the timeline in the top.

Right, I didn't think about that - this will again require a bit of refactoring, as currently TimeGraph assumes it fully occupies the `CaptureWindow`. We should discuss this, it also depends on how large the scrollbar will be in the future (i.e. if it fills the complete height of `CaptureWindow`, or if it stops below the timline.

Let's talk about the next steps before merging this.

karupayun reviewed (commented):
Hi Florian, thanks for this! :) 

I have a question. Did you test what's happening if we set TimeGraph to start at y = 20, instead of 0? I will need this soon because I would like to have the timeline in the top.

florian-kuebler commented on src/OrbitGl/TimeGraph.cpp:
```diff
@@ -530,8 +535,8 @@ void TimeGraph::PrepareBatcherAndUpdatePrimitives(PickingMode picking_mode) {
 
-  text_renderer_static_.PushTranslation(0, -vertical_scrolling_offset_);
-  batcher_.PushTranslation(0, -vertical_scrolling_offset_);
+  // text_renderer_static_.PushTranslation(0, -vertical_scrolling_offset_);
```
Do you want to keep those?

reichlfl replied:
Nope, thanks for noticing

florian-kuebler reviewed (approved):
Cool, thanks!

reichlfl commented:
> Hi Florian, thanks for this! :)
> 
> I have a question. Did you test what's happening if we set TimeGraph to start at y = 20, instead of 0? I will need this soon because I would like to have the timeline in the top.

Edit: I commented on the wrong comment...
So again: Right, I didn't think about this... this will require some more refactoring. We should talk about the next steps here before merging this, maybe I should do this differently then.

reichlfl commented:
I've changed my mind about this after our discussion yesterday, and reverted some of the changes. We still have `VerticalScrollingOffset` in TimeGraph, but this now directly updates the position of its children, so noone besides `TimeGraph` needs to know about this.

florian-kuebler reviewed (approved):
Still looks good to me. Thanks!

Nit for next time: Please avoid squashing/force-pushing.

karupayun reviewed (approved):
LGTM! :)
